cephfs/admin/subvolume.go

```diff
@@ -423,3 +427,129 @@ func (fsa *FSAdmin) UnprotectSubVolumeSnapshot(volume, group, subvolume, name st
 	}
 	return fsa.marshalMgrCommand(m).FilterDeprecated().NoData().End()
 }
+
+// MyFloat is a custom type that implements the MarshalJSON interface.
+// This is used to format float64 values to two decimal places.
+// By default these get converted to integers in the JSON output and
+// fail the command.
+type MyFloat float64
```

Comment:
Maybe call this something like `Float2d` to indicate the 2-digit part? This is exposed to applications that use go-ceph, and a `MyFloat` type is not very descriptive. It would be nicer if there is no need to export a custom float type at all, but I don't immediately see how that can be done without complicating things a lot.

Reply:
I have renamed the type, but I am wondering why it is not un marshalling float properly.